homeassistant/components/schlage/__init__.py

```diff
@@ -0,0 +1,41 @@
+"""The Schlage integration."""
+from __future__ import annotations
+
+from pycognito.exceptions import WarrantException
+import pyschlage
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+from .coordinator import SchlageDataUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.LOCK]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Schlage from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+    try:
+        auth = await hass.async_add_executor_job(
+            pyschlage.Auth, entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD]
+        )
+    except WarrantException as ex:
```

Comment:
If auth fails, `return False` so we don't retry over and over.

In a future PR you can raise ConfigEntryAuthFailed to start a reauth flow

Reply:
Done.

Reply:
Also we should log an error so they know what happened since reauth will likely be next cycle if we merge before beta

Reply:
I added a LOGGER.exception call. Did you have something else in mind?

Reply:
Let’s use error instead of exception since we already know the cause

Reply:
Sounds good. Done. 👍🏻